src/app/app_ui.rs

```diff
@@ -0,0 +1,388 @@
+use std::fmt::Debug;
+
+use eframe::{
+    egui::{
+        text::{CCursor, CCursorRange},
+        *,
+    },
+    App,
+};
+
+use crate::{get_bytes, imageprocessing::*, parse_address};
+
+use super::{check_process_filter, file_helper::*, image_view::ImageView, Application};
+
+#[derive(Debug)]
+enum GetImageError {
+    AddressNotValid,
+    MemoryReadError(std::io::Error),
+}
+
+impl App for Application {
+    fn update(&mut self, ctx: &Context, _frame: &mut eframe::Frame) {
+        CentralPanel::default().show(ctx, |ui| {
+            if self.in_settings {
+                self.draw_settings(ui);
+                if ui.input(|i| i.key_pressed(Key::Escape)) {
+                    self.in_settings = false;
+                }
+            } else {
+                ui.horizontal(|ui| {
+                    ui.vertical(|ui| self.draw_config(ui));
+
+                    ui.separator();
+
+                    ui.vertical(|ui| self.draw_image_block(ui));
+                });
+            }
+        });
+
+        TopBottomPanel::bottom("bottom_panel").show(ctx, |ui| {
+            ui.horizontal(|ui| {
+                self.logger.borrow().show_label(ui);
+                ui.with_layout(Layout::right_to_left(Align::Max), |ui| {
+                    if ui.button("⛭").clicked() {
+                        self.in_settings = !self.in_settings;
+                    }
+                });
+            });
+        });
+    }
+
+    fn save(&mut self, storage: &mut dyn eframe::Storage) {
+        eframe::set_value(storage, eframe::APP_KEY, self);
+    }
+}
+
+impl Application {
+    fn draw_config(&mut self, ui: &mut eframe::egui::Ui) {
+        ui.set_width(200f32);
+        // Process selection
```

Comment:
This function is too large. Consider splitting the drawing logic into multiple functions